design/structure.Rmd

```diff
@@ -0,0 +1,87 @@
+---
+title: "Design for covadj"
+author: "Liming Li"
+output: html_document
+editor_options:
+  chunk_output_type: console
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Objective
+
+The objective is to create a package to include commonly used covariate adjusted analysis method in clinical trials, according to [FDA guidance on adjusting covariates](https://www.fda.gov/regulatory-information/search-fda-guidance-documents/adjusting-covariates-randomized-clinical-trials-drugs-and-biological-products).
+
+# Linear models
+
+Linear models are described in section III.B.
+
+1. Use linear model, include treatment, covariates, intercept with ordinary least square estimator.
+1. Average treatment effect can be robustly obtained.
+1. Robust variance should be used in case of model mis-specification(Huber-White sandwich standard error).
+1. If model includes treatment * covariate interaction, can adopt other robust standard error methods, or using bootstrapping.
+1. Standard error computation best combined with stratification, covariate adjustment and model misspecification.
+
+# Non-linear models
+
+1. Non-collapsibility is a big issue for non-linear models.
+1. HR, OR are usually non-collapsible, while risk difference and RR are collapsible.
+1. CMH is acceptable if conditional treatment effect is of interest.
+1. Covariate adjusted unconditional treatment effect can also be infered, using bootstrapping or statistically reliable methods published.
+1. standardization method is one possible reliable method.
+1. IPTW is also one reliable method.
+1. standard error also best if stratification taken into consideration.
+
+# Implementation of standardization method for linear models
+
+The function name is `lm_std`, and the arguments are quite similar to `lm`.
```

Comment:
General idea: any possibility that we start from `lm` result and then pipe this to a package function to get the covariate adjustment on top?

Reply:
it could be, but we lose some general consistency with other methods, like for standardization methods, we usually need to modify the data and create counter factual treatment and predict the results (sometimes also need to check if the data is of correct structure, like binary treatment) , like iptw we need provide weights obtained from the probability of treatment (other methods we don't include weights); in addition, when it comes to some new methods, then we still need a new interface of the regression. So to make other covariate adjustment methods consistent, it might be good to create some wrappers like this

Reply:
cool. makes sense. Then a consistent prefix with `std_` will be nice

Reply:
I like the idea of the class of method comes first in the prefix if that works for others i.e. 

* std_ <model/method>_  for standisation 
* ipw _ <model/method> _ for inverse weighting ,etc. 

This would follow the [Morris et al.](https://trialsjournal.biomedcentral.com/articles/10.1186/s13063-022-06097-z) overview paper / classification of methods. 

It feels intuitive, especially if autocomplete then helps search within the method class...

Reply:
Hm interesting idea @dgkf ! @clarkliming what do you think?

Reply:
this sounds good but may lead to other confusions, like standardization methods is not actually a linear model, or standardized glm is not a glm; they focus on estimating the treatment effect. so using the `lm` as generic can be a little inappropriate I think; but a similar grammar may be adopted to define the treatment variable (anyway, the treatment effect is the key)